entity-framework/core/modeling/custom-function-mapping.md

```diff
@@ -0,0 +1,202 @@
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Blog.cs#Entity)]
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Post.cs#Entity)]
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Tag.cs#Entity)]
+
+And the following model configuration:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/BloggingContext.cs#EntityConfiguration)]
+
+Blog can have many posts, each post can be tagged with multiple tags, and each tag can be associated with multiple posts: many-to-many relationship.
+
+Next, create the user-defined function `DistinctTagsCountForBlogPosts`, which returns the count of unique tags associated with all the posts of a given blog, based on the blog `Id`:
+
+```sql
+CREATE FUNCTION dbo.DistinctTagsCountForBlogPosts(@id int)
+    RETURNS int
+    AS
+    BEGIN
+        RETURN (SELECT COUNT(*) FROM(
+            SELECT DISTINCT t.TagId FROM dbo.Tags AS t
+            JOIN dbo.PostTag AS pt ON t.TagId = pt.TagId
+            JOIN dbo.Posts AS p ON p.PostId = pt.PostId
+            JOIN dbo.Blogs AS b ON b.BlogId = p.BlogId
+            WHERE b.BlogId = @id) AS subquery);
+    END
+```
+
+To use it in EF Core, define the following CLR method, which we will map to the user-defined function:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/BloggingContext.cs#BasicFunctionDefinition)]
+
+In the example, the method is defined on `DbContext`, but it can also be defined in other places.
+
+> [!NOTE]
+> Body of the CLR method is not important. EF Core only looks at the method signature.
+
+This function definition can be associated with user-defined function in the model configuration:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/BloggingContext.cs#BasicFunctionConfiguration)]
+
+> [!NOTE]
+> By default EF Core tries to map CLR function to a user-defined function with the same name. If the names are different, we can use `HasName` to select the correct name for the user-defined function we want to map to.
+
+Now, executing the following query:
+
+[!code-csharp[Main](../../../samples/core/Modeling/CustomFunctionMapping/Program.cs#BasicQuery)]
+
+Will produce this SQL:
+
+```sql
+SELECT [b].[BlogId], [b].[Rating], [b].[Url]
+FROM [Blogs] AS [b]
+WHERE [dbo].[DistinctTagsCountForBlogPosts]([b].[BlogId]) > 2
+```
+
+## Mapping method to function defined in the model
+
+EF Core also allows for user-defined functions that don't map to the corresponding function in the database. It can be done by specifying the function body using the [Microsoft.EntityFrameworkCore.Query.SqlExpressions](dotnet/api/microsoft.entityframeworkcore.query.sqlexpressions) API. Function body is provided using `HasTranslation` method during user-defined function configuration.
+
+In the example below, we'll create a function that computes difference between two integers.
```

Comment:
The function thing bothered me as well (it's too complicated) but I also didn't want to do something like "addition" because that would be even more meaningless, will try to think of something.